Src/FluentAssertions/Common/TypeMemberReflector.cs

```diff
@@ -18,7 +18,7 @@ public TypeMemberReflector(Type typeToReflect, MemberVisibility visibility)
     {
         Properties = LoadProperties(typeToReflect, visibility);
         Fields = LoadFields(typeToReflect, visibility);
-        Members = Properties.Concat<MemberInfo>(Fields).ToArray();
+        Members = [.. Properties, .. Fields];
```

Comment:
According to ILSpy this is lowered to
```cs
int num = 0;
MemberInfo[] array = new MemberInfo[properties.Length + fields.Length];
PropertyInfo[] array2 = properties;
foreach (PropertyInfo propertyInfo in array2)
{
    array[num] = propertyInfo;
    num++;
}
FieldInfo[] array3 = fields;
foreach (FieldInfo fieldInfo in array3)
{
    array[num] = fieldInfo;
    num++;
}
Members = array;
```

According to SharpLab it's lowered to this 
(haven't checked if the difference it because of some API only available in .NET 8)
```cs
int num = 0;
MemberInfo[] array = new MemberInfo[a.Length + b.Length];
Span<MemberInfo> span = new Span<MemberInfo>(a);
span.CopyTo(new Span<MemberInfo>(array).Slice(num, span.Length));
num += span.Length;
Span<MemberInfo> span2 = new Span<MemberInfo>(b);
span2.CopyTo(new Span<MemberInfo>(array).Slice(num, span2.Length));
num += span2.Length;
return array;
```

[SharpLab](https://sharplab.io/#v2:EYLgtghglgdgPgAQAwAIEEYB0AZWBHAbgFgAoZNLAJQFMAzAG2oGMAXKAexmJNIwDYUAWWphg1AE4BJGLXYBtALooAwpyYQWACmGiJ02YpQQANEJFipM+UuABKFAF4AfEcyqY6rXcwAVdgEFxcQgAT01bbl50AR0LfWsUAGUAB3FqCAATbXM9K0MTM11LAxt7ZxQ5TEwCquAFAiA)

This shouldn't matter for non-netfx as `Enumerable.Concat` has since https://github.com/dotnet/corefx/pull/11492 (.NET Core 2.0) checked if it could fetch collection sizes to pre-allocate the resulting array.
Current implemention of [Concat2Iterator.<TSource>ToArray()](https://source.dot.net/#System.Linq/System/Linq/Concat.SpeedOpt.cs,61)